### PR TITLE
feature/Core of wallet management features-#22

### PR DESCRIPTION
**Pull request by rpajdak**

### CODE REVIEW

**Review by dawciobiel — CHANGES_REQUESTED** [review]
A request to solve a task in accordance with its requirements only. Description:
#22

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@RestController
+@RequiredArgsConstructor
+@Slf4j
+class WalletRestController {
+
+  @NonNull
+  private final WalletServiceImpl walletService;
+
+  @PostMapping("api/v1/wallets")
+  public ResponseEntity<WalletDto> create(@Valid @RequestBody CreateWalletDto createWalletDto) {
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java:27]: The task was going on create empty class with `@RestController` annotation

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/Wallet.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @EqualsAndHashCode.Include
+  @Column(updatable = false)
+  private Long id;
+
+  @ToString.Include
+  @Column(nullable = false)
+  private UUID walletId;
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/Wallet.java:23]: I don't get reason, why u implemented second ID here. If you look up to issue you can see following -> 
"The wallet entity should contain:
Id,
wallet name,
wallet creation date.
Wallet name:"

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/dto/CreateWalletDto.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+import javax.validation.constraints.Size;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/dto/CreateWalletDto.java:12]: There wasn't any mention about dto

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletService.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import pl.byczazagroda.trackexpensesappbackend.model.dto.CreateWalletDto;
+import pl.byczazagroda.trackexpensesappbackend.model.dto.WalletDto;
+
+public interface WalletService {
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletService.java:6]: If you look up to issue, you can see statement about interface -> "Service interface without method declarations"

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class WalletServiceImpl implements WalletService {
+
+  private final WalletRepository walletRepository;
+
+  @Override
+  public WalletDto createWallet(CreateWalletDto createWalletDto) {
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java:20]: First of all, I've no idea why u've used builder here. Second thing that it was going about creating empty service, and it's implementation.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  @MockBean
+  WalletServiceImpl walletService;
+
+  @SneakyThrows
+  @Test
+  void shouldReturnCreatedCodeCodeWhenWalletCreated() {
+
+    CreateWalletDto createWalletDto = new CreateWalletDto("Test wallet name");
+
+    mockMvc.perform(post("/api/v1/wallets")
```
Comment (MateuszBednarczyk) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestControllerTest.java:32]: I think, that you should create empty tests (same as service, implementation and controller) which's always successful

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  WalletServiceImpl underTest;
+
+  @Mock
+  WalletRepository walletRepository;
+
+  @Test
+  void shouldCreateWallet(){
+    //given
+
+    //when
```
Comment (MateuszBednarczyk) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceTest.java:25]: test is actually correct, but we didn't mention about this method.

**Review by MateuszBednarczyk — CHANGES_REQUESTED** [review]

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java (hunk 1 above)
Comment (rpajdak) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java:27]: No, it was not specified in the task.
image

Comment (rpajdak) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java:27]: Whats the point of making empty class?

Comment (MateuszBednarczyk) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletRestController.java:27]: The point, is the task was about creating app core/structure

**Pull request closed without merging**